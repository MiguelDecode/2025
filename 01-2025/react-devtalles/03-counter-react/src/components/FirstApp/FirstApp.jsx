// import "./FirstApp.css";
import PropTypes from "prop-types";

export const FirstApp = ({
  title = "Fernando Herrera",
  subTitle = "No hay subtÃ­tulo",
  name,
}) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number,
};
