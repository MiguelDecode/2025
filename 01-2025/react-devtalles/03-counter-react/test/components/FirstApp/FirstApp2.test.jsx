import { render, screen } from "@testing-library/react";
import { FirstApp } from "../../../src/components/FirstApp/FirstApp";

describe("Tests on <FirstApp/>", () => {
  const title = "Hola, soy Goku";
  const subtitle = "Soy un subtÃ­tulo";

  test("should must match with the snapshot", () => {
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test('should show the message "Hola, soy Goku"', () => {
    render(<FirstApp title={title} />);

    expect(screen.getByText(title)).toBeTruthy();
  });

  test("should show the title inside a h1", () => {
    render(<FirstApp title={title} />);

    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("should show a subtitle send as a prop", () => {
    render(<FirstApp title={title} subTitle={subtitle} />);

    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
