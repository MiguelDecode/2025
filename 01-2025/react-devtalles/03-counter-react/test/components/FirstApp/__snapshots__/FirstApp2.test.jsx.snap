// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <FirstApp/> should must match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p />
</div>
`;
